src/hooks/useContract.ts

```diff
@@ -141,17 +145,20 @@ export function usePoolRegistry(): PoolRegistry | null {
   }, [masterRegistryContract])
 
   return useMemo(() => {
-    if (!library || !contractAddress) return null
+    if (!contractAddress) return null
     return getContract(
       contractAddress,
       POOL_REGISTRY_ABI,
-      library,
+      signer || provider,
     ) as PoolRegistry
-  }, [contractAddress, library])
+  }, [contractAddress, provider, signer])
 }
 
 export function usePoolRegistryMultiCall(): MulticallContract<PoolRegistry> | null {
-  const { library } = useActiveWeb3React()
+  const { data: signer } = useSigner()
+  const provider = useProvider()
+  const signerOrProvider = signer || provider
```

Comment:
Why not use `signerOrProvider` from `useActiveWeb3React`?

Reply:
While debugging, I used `signer` and `provider` separately. We can use`signerOrProvider`.